src/commitlint/cli.py

```diff
@@ -119,50 +126,59 @@ def _get_commit_message_from_file(filepath: str) -> str:
         return commit_message
 
 
-def _handle_commit_message(commit_message: str, skip_detail: bool) -> None:
+def _handle_commit_message(
+    commit_message: str, skip_detail: bool, quiet: bool = False
+) -> None:
     """
     Handles a single commit message, checks its validity, and prints the result.
 
     Args:
         commit_message (str): The commit message to be handled.
         skip_detail (bool): Whether to skip the detailed error linting.
+        quiet (bool): Whether to print the detailed errors
 
     Raises:
         SystemExit: If the commit message is invalid.
     """
     success, errors = lint_commit_message(commit_message, skip_detail=skip_detail)
 
-    if success:
-        sys.stdout.write(f"{VALIDATION_SUCCESSFUL}\n")
-    else:
-        _show_errors(commit_message, errors, skip_detail=skip_detail)
+    if quiet and errors:
         sys.exit(1)
 
+    elif not quiet:
+        if success:
+            sys.stdout.write(f"{VALIDATION_SUCCESSFUL}\n")
+        else:
+            _show_errors(commit_message, errors, skip_detail=skip_detail)
+            sys.exit(1)
+
 
 def _handle_multiple_commit_messages(
-    commit_messages: List[str], skip_detail: bool
+    commit_messages: List[str], skip_detail: bool, quiet: bool = False
 ) -> None:
     """
     Handles multiple commit messages, checks their validity, and prints the result.
 
     Args:
         commit_messages (List[str]): List of commit messages to be handled.
         skip_detail (bool): Whether to skip the detailed error linting.
-
+        quiet (bool): Whether to show the error and messages in console
     Raises:
         SystemExit: If any of the commit messages is invalid.
     """
+    # if not quiet:
     has_error = False
     for commit_message in commit_messages:
         success, errors = lint_commit_message(commit_message, skip_detail=skip_detail)
-        if not success:
+        if not success and not quiet:
             has_error = True
             _show_errors(commit_message, errors, skip_detail=skip_detail)
             sys.stderr.write("\n")
 
     if has_error:
         sys.exit(1)
-    else:
+
+    elif not has_error and not quiet:
```

Comment:
suggestion: If we are using `elif` there is no use of `not has_error` condition.